File: client/src/types/triage.ts
export interface SOAPNote {
    subjective: string;
    objective: string;
    assessment: string;
    plan: string;
}

export interface VitalSigns {
    temperature: number;
    blood_pressure_systolic: number;
    blood_pressure_diastolic: number;
    heart_rate: number;
    respiratory_rate: number;
    oxygen_saturation: number;
    recorded_at: string;
    recorded_by: string;
}

export interface TriageRecord {
    id: string;
    patient_id: string;
    audio_file_url: string;
    language: string;
    transcription: string;
    translation: string;
    soap_note?: SOAPNote | null;
    vitals?: VitalSigns | null;
    risk_score: number;
    triage_tier: 'EMERGENCY' | 'URGENT' | 'SEMI_URGENT' | 'ROUTINE';
    preliminary_zone?: 'EMERGENCY' | 'URGENT' | 'SEMI_URGENT' | 'ROUTINE' | 'STABLE' | 'ABNORMAL' | null; // vitals-only fast estimate
    vitals_status?: 'STABLE' | 'ABNORMAL';
    preliminary_precautions?: string[];
    specialty: string;
    patient_age?: number | null;
    status: 'pending' | 'in_progress' | 'ready_for_review' | 'finalized' | 'exported' | 'failed';
    created_at: string;
    updated_at: string;
}

File: server/models/triage.py
from datetime import datetime
from typing import List, Optional
from pydantic import BaseModel

class AcousticAnomaly(BaseModel):
    type: str  # 'respiratory_distress' | 'cough' | 'voice_strain' | 'wheezing'
    confidence: float
    timestamp: float

class SOAPNote(BaseModel):
    subjective: str
    objective: str
    assessment: str
    plan: str

class VitalSigns(BaseModel):
    temperature: float  # Celsius
    blood_pressure_systolic: int  # mmHg
    blood_pressure_diastolic: int  # mmHg
    heart_rate: int  # bpm
    respiratory_rate: int  # breaths per minute
    oxygen_saturation: int  # percentage
    recorded_at: datetime
    recorded_by: str

class TriageRecord(BaseModel):
    id: str
    patient_id: str
    audio_file_url: str
    language: str
    transcription: str = ""
    translation: str = ""
    soap_note: Optional[SOAPNote] = None
    vitals: Optional[VitalSigns] = None
    risk_score: int = 0
    triage_tier: str = "ROUTINE" # 'EMERGENCY' | 'URGENT' | 'SEMI_URGENT' | 'ROUTINE'
    specialty: str = "General Medicine"
    status: str  # 'pending' | 'in_progress' | 'ready_for_review' | 'finalized'
    is_seen: bool = False
    created_at: datetime
    updated_at: datetime
